e2e/specs/register-new-patient.spec.ts

```diff
@@ -31,29 +30,38 @@ test('Register a new patient', async ({ page, api }) => {
     await patientRegistrationPage.waitUntilTheFormIsLoaded();
   });
 
-  await test.step('And then I click on fill new values into the registration form and then click the `Submit` button', async () => {
+  await test.step('And then I fill the registration form and then click the `Submit` button', async () => {
     await patientRegistrationPage.fillPatientRegistrationForm(formValues);
+    await expect(page.getByText(/new patient created/i)).toBeVisible();
+  });
+
+  await test.step("Then I should be redirected to the new patient's chart page", async () => {
+    const patientChartUrlRegex = new RegExp('^[\\w\\d:\\/.-]+\\/patient\\/[\\w\\d-]+\\/chart\\/.*$');
+    await page.waitForURL(patientChartUrlRegex);
+    await expect(page).toHaveURL(patientChartUrlRegex);
   });
 
-  await test.step("Then I should be redirected to the new patient's chart page and a new patient record should be created from the information captured in the form", async () => {
-    await expect(page).toHaveURL(new RegExp('^[\\w\\d:\\/.-]+\\/patient\\/[\\w\\d-]+\\/chart\\/.*$'));
-    const patientUuid = /patient\/(.+)\/chart/.exec(page.url())?.[1] ?? null;
-    expect(patientUuid).not.toBeNull();
-
-    const patient = await getPatient(api, patientUuid);
-    const { person } = patient;
-    const { givenName, middleName, familyName, sex } = formValues;
-
-    expect(person.display).toBe(`${givenName} ${middleName} ${familyName}`);
-    expect(person.gender).toMatch(new RegExp(sex[0], 'i'));
-    expect(dayjs(person.birthdate).format('DD/MM/YYYY')).toBe(
-      `${formValues.birthdate.day}/${formValues.birthdate.month}/${formValues.birthdate.year}`,
-    );
-    expect(person.preferredAddress.address1).toBe(formValues.address1);
-    expect(person.preferredAddress.cityVillage).toBe(formValues.cityVillage);
-    expect(person.preferredAddress.stateProvince).toBe(formValues.stateProvince);
-    expect(person.preferredAddress.country).toBe(formValues.country);
-    expect(person.attributes[0].display).toBe(formValues.phone);
+  await test.step("And I should the newly registered patient's details displayed in the patient banner", async () => {
+    const patientBanner = page.locator('header[aria-label="patient banner"]');
+    await patientBanner.waitFor({ state: 'visible' });
+
+    await expect(page.getByLabel('patient banner').getByText(/johnny donny ronny/i)).toBeVisible();
```

Comment:
Using `patientBanner.getBy..` should work. plus https://github.com/openmrs/openmrs-esm-patient-management/pull/1286#discussion_r1721733173

```suggestion
    await expect(patientBanner.getByText("Johnny Donny Ronny")).toBeVisible();
```